runtime/include/chpl-comm.h

```diff
@@ -77,6 +77,44 @@ extern const int chpl_heterogeneous;
 // chpl_comm_nb_handle_t must be defined in the comm layer header
 // chpl-comm-task-decls.h
 
+// uses comm-layer specific chpl_comm_bundleData_t
+// uses task-layer specific chpl_task_bundleData_t
+typedef struct {
+  chpl_task_bundle_t task_bundle;
+  chpl_fn_int_t fid;
+  chpl_comm_bundleData_t comm;
+  // arguments immediately follow
+} chpl_comm_on_bundle_t;
+
+typedef chpl_comm_on_bundle_t *chpl_comm_on_bundle_p;
+
+static inline
+chpl_task_bundle_t* chpl_comm_on_bundle_task_bundle(chpl_comm_on_bundle_t* a)
+{
+  return &a->task_bundle;
+}
+
+//
+// Call a chpl_ftable[] function in a task.
+//
+// This is a convenience function for use by the module code, in which
+// we have function table indices rather than function pointers.
+//
+static inline
+void chpl_task_taskCallFTable(chpl_fn_int_t fid,      // ftable[] entry to call
```

Comment:
Let's not define chpl_task_*() functions in chpl-comm.h.  Instead, leave chpl_task_taskCallFTable() where it was in chpl-tasks.h and define it as taking a chpl_task_bundle_p.  Add a new chpl_comm_taskCallFtable() that takes a chpl_comm_on_bundle_p here, call it from the locale model module code, and have it in turn call chpl_task_taskCallFTable() with the chpl_comm_on_bundle_task_bundle().  If everything is static inline it'll all collapse down to the same amount of code, but it'll be easier to find everything.